### DISCUSSION TITLE
Local env, when visiting 'localhost:3000', it turns to 404 page. So how can I redirect to home page?

### DISCUSSION THREAD

**Discussion by ccloveyy** [nuxt/nuxt#17667] [Questions]
This is the first time I've tried nuxt and I really need some help.

http://localhost:3000 => 404 page
http://localhost:3000/home => home page

I wanna redirect to home page when visiting http://localhost:3000. What should I do?

Here I've tried to use middleware：
middleware/redirect.js:

```javascript
export default defineNuxtRouteMiddleware((to, from) => {
  console.log('to:',to);  // print nothing
  if (to.path === '/') {
    return navigateTo('/home')
  }
})
```

app.vue:
```html
<template>
  <div>
    <NuxtLayout name="starve">
      <NuxtPage/>
    </NuxtLayout>
  </div>
</template>

<script setup>
definePageMeta({
    middleware: ['redirect']
})
</script>
```

**Comment by zhoutengshen** [comment]
oh ,i have a same problem
i got a idel from [Add `useRedirect` as a core composable](https://github.com/nuxt/framework/discussions/3576)
maybe be can help you
